# Project Capstone: Fraud Detection on Auto Insurance Claims
**Notebook 4: Fraud Claim Analysis, Conclusion & Recommendations**

## TABLE OF CONTENTS

**1. Understanding Data** <br>
**2. Preprocessing** <br>
**3. Feature Engineering, Modelling, Feature Selection, Hyperparameter Tuning and Final Model** <br>
**4. Fraud Claim Analysis, Conclusion & Recommendations (This Notebook)** <br>
- [Fraud Claim Analysis](#Fraud-Claim-Analysis) <br>
- [Conclusion](#Conclusion) <br>
- [Recommendations](#Recommendations) <br>
- [Tableau Dashboard](#Tableau-Dashboard) <br>

### Import Libraries

In [1]:
# Import library
import pandas as pd

### Import Data

In [2]:
# Read data from csv file.
auto_insurance_dataframe = pd.read_csv("./datasets/auto_insurance_claims_data.csv")

# Increase display range for easier viewing and EDA.
pd.options.display.max_columns = None
pd.options.display.max_rows = None

In [3]:
### Visualize Data

In [4]:
# Print number of rows and columns in dataframe.
print(f"Shape:", auto_insurance_dataframe.shape)

Shape: (1000, 40)


In [5]:
# Display first 3 rows of dataframe.
auto_insurance_dataframe.head(3)

months_as_customer  age  policy_number policy_bind_date policy_state  \
0                 328   48         521585       2014-10-17           OH   
1                 228   42         342868       2006-06-27           IN   
2                 134   29         687698       2000-09-06           OH   

  policy_csl  policy_deductable  policy_annual_premium  umbrella_limit  \
0    250/500               1000                1406.91               0   
1    250/500               2000                1197.22         5000000   
2    100/300               2000                1413.14         5000000   

   insured_zip insured_sex insured_education_level insured_occupation  \
0       466132        MALE                      MD       craft-repair   
1       468176        MALE                      MD  machine-op-inspct   
2       430632      FEMALE                     PhD              sales   

  insured_hobbies insured_relationship  capital-gains  capital-loss  \
0        sleeping              husband          53300             0   
1         reading       other-relative              0             0   
2     board-games            own-child          35100             0   

  incident_date             incident_type  collision_type incident_severity  \
0    2015-01-25  Single Vehicle Collision  Side Collision      Major Damage   
1    2015-01-21             Vehicle Theft               ?      Minor Damage   
2    2015-02-22   Multi-vehicle Collision  Rear Collision      Minor Damage   

  authorities_contacted incident_state incident_city  incident_location  \
0                Police             SC      Columbus     9935 4th Drive   
1                Police             VA     Riverwood       6608 MLK Hwy   
2                Police             NY      Columbus  7121 Francis Lane   

   incident_hour_of_the_day  number_of_vehicles_involved property_damage  \
0                         5                            1             YES   
1                         8                            1               ?   
2                         7                            3              NO   

   bodily_injuries  witnesses police_report_available  total_claim_amount  \
0                1          2                     YES               71610   
1                0          0                       ?                5070   
2                2          3                      NO               34650   

   injury_claim  property_claim  vehicle_claim auto_make auto_model  \
0          6510           13020          52080      Saab        92x   
1           780             780           3510  Mercedes       E400   
2          7700            3850          23100     Dodge        RAM   

   auto_year fraud_reported  _c39  
0       2004              Y   NaN  
1       2007              Y   NaN  
2       2007              N   NaN

# Fraud Claim Analysis

## Policy Annual Premium

In this section, policy annual premium will be analyzed in details and by different features.

### Total Policy Annual Premium and Total Policy Count

In [6]:
# Calculate total policy annual premium.
policy_annual_premium = auto_insurance_dataframe["policy_annual_premium"].sum()

# Print total policy annual premium.
print("Policy Annual Premium: $" + '{:,}'.format(round(policy_annual_premium, )))

Policy Annual Premium: $1,256,406


In [7]:
# Calculate total policy count.
policy_count = auto_insurance_dataframe["policy_number"].count()

# Print total policy count.
print("Policy Count: " + '{:,}'.format(round(policy_count, )))

Policy Count: 1,000


In [8]:
# Calculate premium per policy.
premium_per_policy = policy_annual_premium / policy_count

# Print total premium per policy.
print("Annual Premium Per Policy: $" + '{:,}'.format(round(premium_per_policy, )))

Annual Premium Per Policy: $1,256


### Policy Annual Premium and Policy Count Breakdown By Gender

![image.png](./images/tableau_policy_by_premium_by_count_gender.png)
![image.png](./images/tableau_annual_premium_per_policy_count_by_gender.png)

### Policy Annual Premium and Policy Count Breakdown By Policy Issuance State

![image.png](./images/tableau_policy_annual_premium_policy_count_by_state.png)
![image.png](./images/tableau_policy_annual_premium_per_policy_count_by_state.png)

### Policy Annual Premium and Policy Count Breakdown By Policy Bind Year

![image.png](./images/tableau_premium_policy_annual_premium_policy_count_policy_bind_year.png)
![image.png](./images/tableau_premium_per_policy_by_policy_bind_year.png)

### Policy Annual Premium and Policy Count Breakdown By Insured's Occupation

![image.png](./images/tableau_policy_annual_premium_count_by_insured_occupation.png)
![image.png](./images/tableau_policy_annual_premium_per_policy_count_by_insured_occupation.png)

**Observations**:
- Total policy premium collected from 1,000 policies was \$1,256,406 with premium per policy at about \\$1,256 (rounded to nearest dollar amount).
- Policy Annual Premium and Policy Count Breakdown By Gender
    - Between 2 gender, policy count and policy premium by female were higher than that of male. It suggests that female bought more policies and paid more policy premium than that of male.
    - However, annual policy premium per policy by male was \$1,267 , which was higher than that of female at \\$1,248. It suggests that, in each policy, male paid higher premium than that of female.
- Policy Annual Premium and Policy Count Breakdown By State
    - Policy annual premium, policy count and policy annual premium per policy by Ohio were \$443,674, 352 and \\$1,260 respectively, which were the highest among all 3 states.
- Policy Annual Premium and Policy Count Breakdown By Policy Bind Year
    - Among all policy bind years, policy bind year 1991 had the highest policy annual premium, policy count and annual premium per policy count at \\$72,485, 55 and \\$1,318 respectively. In contrast, policy bind year 2015 had the lowest policy annual premium and policy count at \$3,728 and 3 respectively; and policy bind year 2005 had the lowest annual premium per policy count at \\$1,204.
- Policy Annual Premium and Policy Count Breakdown By Insured's Occupation
    - Among all insured's occupations, machine-op-inspct had the highest policy annual premium and policy count at \\$113,681 and 93 respectively. In contrast, farming-fishing had the lowest policy annual premium and policy count at \$67,773 and 53 respectively. 
    - Nonetheless, insured's occupation: other-service, had the highest policy annual premium per policy count at \\$1,307.

## Claim

In this section, claim will be analyzed in details and by different features.

### Total Claim Amount and Total Claim Count

In [9]:
# Calculate total claim amount.
total_claim_amount = auto_insurance_dataframe["total_claim_amount"].sum()

# Print total claim amount.
print("Total Claim Amount: $" + '{:,}'.format(round(total_claim_amount, )))

Total Claim Amount: $52,761,940


In [10]:
# Calculate total claim count.
claim_count = auto_insurance_dataframe["total_claim_amount"].count()

# Print total ciam count.
print("Total Claim Count: " + '{:,}'.format(round(claim_count, )))

Total Claim Count: 1,000


In [11]:
# Calculate premium per policy.
claim_per_policy = total_claim_amount / claim_count

# Print total premium per policy.
print("Claim Per Policy: $" + '{:,}'.format(round(claim_per_policy, )))

Claim Per Policy: $52,762


**Observation**:
- Total claim amount from 1,000 policies was \$52,761,940 with claim per policy at about \\$52,762 (rounded to nearest dollar amount).

### Total Claim Amount and Total Claim Count Breakdown By Gender

![image.png](./images/tableau_claim_amount_claim_count_by_gender.png)
![image.png](./images/tableau_claim_amount_per_policy_count_by_gender.png)

### Total Claim Amount and Total Claim Count Breakdown By Policy Issuance State

![image.png](./images/tableau_total_claim_amount_claim_count_by_policy_issuance_state.png)
![image.png](./images/tableau_total_claim_amount_per_policy_count_by_state.png)

### Total Claim Amount and Total Claim Count Breakdown By Policy Bind Year

![image.png](./images/tableau_total_claim_amount_claim_count_policy_bind_year.png)
![image.png](./images/tableau_total_claim_amount_per_policy_count_by_policy_bind_year.png)

### Total Claim Amount and Total Claim Count Breakdown By Insured's Occupation

![image.png](./images/tableau_total_claim_amount_claim_count_by_insured_occupation.png)
![image.png](./images/tableau_total_claim_amount_per_claim_count_by_insured_occupation.png)

**Observations**:
- Total claim amount paid in 1,000 policies was \$52,761,940 with claim amount per policy at about \\$52,762 (rounded to nearest dollar amount).
- Total Claim Amount and Total Claim Count Breakdown By Gender
    - Between 2 gender, total amount amount, total claim count and total claim amount per policy incurred by female were higher than that of male. 
- Total Claim Amount and Total Claim Count Breakdown By Policy Issuance State
    - Policy annual premium and policy count by Ohio were \$18,468,450 and 352 respectively, which were the highest among all 3 states. Nonetheless, Indiana had highest total claim amount per policy count at \\$53,007.
- Total Claim Amount and Total Claim Count Breakdown By Policy Bind Year
    - Among all policy bind years, policy bind year 1990, 1991 and 1993 had the highest total claim amount, total claim count and total claim amount per policy count at \\$2,686,070, 55 and \\$60,799 respectively.
- Total Claim Amount and Total Claim Count Breakdown By Insured's Occupation
    - Among all insured's occupations, machine-op-inspct had the highest total claim amount and total claim count at \\$5,041,460 and 93 respectively. In contrast, farming-fishing had the lowest total claim amount and total claim count at \$2,637,760 and 53 respectively.
    - Nonetheless, insured's occupation: handlers-cleaners, had the highest total claim amount per policy count at \\$61,658.

## Claim Ratio

In this section, claim ratio will be analyzed in details and by different features.

![image.png](./images/tableau_overall_claim_ratio.png)
![image.png](./images/tableau_claim_ratio_gender.png)
![image.png](./images/tableau_claim_ratio_policy_issuance_state.png)
![image.png](./images/tableau_claim_ratio_policy_bind_year.png)
![image.png](./images/tableau_claim_ratio_Insured_occupation.png)
![image.png](./images/tableau_claim_ratio_Insured_education.png)
![image.png](./images/tableau_claim_ratio_Insured_hobbies.png)
![image.png](./images/tableau_claim_ratio_Insured_incident_type.png)
![image.png](./images/tableau_claim_ratio_Insured_incident_severity.png)
![image.png](./images/tableau_claim_ratio_Insured_authorities_contacted.png)
![image.png](./images/tableau_claim_ratio_Insured_police_report.png)
![image.png](./images/tableau_claim_ratio_Insured_property_damage.png)
![image.png](./images/tableau_claim_ratio_collision_type.png)
![image.png](./images/tableau_claim_ratio_car_model.png)
![image.png](./images/tableau_claim_ratio_incident_city.png)
![image.png](./images/tableau_claim_ratio_relationship_to_insured.png)

**Observations**:
- Overall claim ratio was very high at 4199\%. It means total claims amount was about 42 times of total premium collected.
- Female had higher claim ratio at 4276\% compared to that of male at 4112\%.
- Indiana had the highest claim ratio at 4222\% among 3 states.
- Among all policy bind years, policy bind year 2005 had the highest claim ratio at 4841\%.
- Among all insured's occupation, handlers-cleaners had the highest claim ratio at 4772\%.
- Among all insured's education, Masters had the highest claim ratio at 4433\%.
- Among all insured's hobbies, cross-fit had the highest claim ratio at 4843\%.
- Among all incident types, single vehicle collision had the highest claim ratio at 5057\%.
- Among all incident severity, major damage had the highest claim ratio at 5110\%.
- Among all authorities contacted, others had the highest claim ratio at 5184\%.
- Among all police report availability options, ? had the highest claim ratio at 4334\%.
- Among all property damage status, "Yes" had the highest claim ratio at 4291\%.
- Among all collision types, front collision had the highest claim ratio at 5188\%.
- Among all car models, Escape had the highest claim ratio at 5366\%.
- Among all incident cities, Columbus had the highest claim ratio at 4409\%.
- Among all relationship to insured, not in family had the highest claim ratio at 4276\%.

## Fraud

In this section, fraud will be analyzed in details and by different features.

### Fraud Overview

![image.png](./images/tableau_fraud_overview.png)

### Fraud Breakdown By Insured's Hobbies

![image.png](./images/tableau_fraud_breakdown_by_insured_hobbies.png)
![image.png](./images/tableau_fraud_ratio_breakdown_by_insured_hobbies.png)

### Fraud Breakdown By Incident Type

![image.png](./images/tableau_fraud_breakdown_by_incident_type.png)
![image.png](./images/tableau_fraud_ratio_breakdown_by_incident_type.png)

### Fraud Breakdown By Incident Severity

![image.png](./images/tableau_fraud_breakdown_by_incident_severity.png)
![image.png](./images/tableau_fraud_ratio_breakdown_by_incident_severity.png)

### Fraud Breakdown By Authorities Contacted

![image.png](./images/tableau_fraud_breakdown_by_authorities_contacted.png)
![image.png](./images/tableau_fraud_ratio_breakdown_by_authorities_contacted.png)

### Fraud Breakdown By Property Damage

![image.png](./images/tableau_fraud_breakdown_by_property_damage.png)
![image.png](./images/tableau_fraud_ratio_breakdown_by_property_damage.png)

**Observations**:
- Although number of fraudulent insurance claims was 247, claim ratio was significantly at 4823%. It means total claim amount from fraudulent insurance claims was 48 times of total premium collected from the same number of policies.
- Insureds, whose hobbies were chess or cross-fit, were highly likely to commit fraud, with fraud ratio at 83% and 74% respectively.
- For Incidents types vehicle theft or incident type parked car, as their fraud ratio were at 9% and 10% respectively, these incidents types were not likely to be fraudulent.
- For Incidents severity minor damage or total loss or trivial damage, as their fraud ratio were at 11%, 13% and 7% respectively, incidents with these severity were not likely to be fraudulent.
- For no property damage, as fraud ratio was at %, incidents with no property damage were not likely to be fraudulent.

Summary of fraud reported by claim count, overall claim fraud ratio and fraud ratio by insured's hobbies is this [Tableau Dashboard](https://public.tableau.com/app/profile/gerald.chin/viz/fraud_detection_on_auto_insurance_claim/fraud_detection_on_auto_insurance_claim_overview_of_fraud).

# Conclusion

Gaussian Naive Bayes is chosen as the final model as Gaussian Naive Bayes 
- is not overfitting;
- has highest testing ROC AUC at 86%;
- has high f1 score at 74%; and
- can do better classification based on small dataset. ([source](https://www.educba.com/naive-bayes-vs-logistic-regression)).

This model helps to predict fraud and reduce loss from fraud claims, which translates to saving.

# Recommendations

These are the limitations observed:
- dataset is small, with data from 1 Jan 2015 to 1 March 2015;
- results generalizability limited to Ohio, Illinois and Indiana;
- no description for certain columns such as _c39, policy_csl and insured_relationship;
- no information on the source of data, whether it is from one insurance company, a few companies or more; and
- Naïve Bayes model do not work well if features are correlated.

Hence, it is recommended to 
- seek clarification on data source and data description; and
- continuous training and improving model with more data.